presto-raptor/src/test/java/com/facebook/presto/raptor/TestRaptorIntegrationSmokeTest.java

```diff
@@ -130,6 +136,135 @@ public void testBucketNumberHiddenColumn()
     }
 
     @Test
+    public void testShardingByTemporalDateColumn()
+            throws Exception
+    {
+        assertUpdate("CREATE TABLE test_shard_temporal_date " +
+                        "WITH (TEMPORAL_COLUMN='orderdate') AS " +
+                        "SELECT orderdate, orderkey " +
+                        "FROM orders " +
+                        "WHERE orderdate < date '1992-02-08'",
```

Comment:
Why do these tests filter by date? Is it too slow to use the whole table?

My concern is that these tests would pass if there were no matching rows (for example, if you picked `1990` instead of `1992`). We should add something at the top to validate there are multiple dates:
```java
assertTrue(getOnlyElement(computeActual("SELECT count(DISTINCT orderdate) >= 2 FROM orders WHERE orderdate < date '1992-02-08'").getOnlyColumnAsSet()));